mule-extensions-api/src/main/java/org/mule/runtime/extension/api/manifest/ImmutableExtensionManifest.java

```diff
@@ -0,0 +1,73 @@
+/*
+ * Copyright (c) MuleSoft, Inc.  All rights reserved.  http://www.mulesoft.com
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+package org.mule.runtime.extension.api.manifest;
+
+/**
+ * Immutable implementation of {@link ExtensionManifest}
+ *
+ * @since 1.0
+ */
+final class ImmutableExtensionManifest implements ExtensionManifest
+{
+
+    private final String name;
+    private final String description;
+    private final String version;
+    private final DescriberManifest describerManifest;
+
+    /**
+     * Creates a new instance
+     *
+     * @param name              the extension's name
+     * @param description       the extension's description
+     * @param version           the extension's version
+     * @param describerManifest the extension's {@link DescriberManifest}
+     */
+    ImmutableExtensionManifest(String name, String description, String version, DescriberManifest describerManifest)
```

Comment:
Argument validation must be moved form the builders to the corresponding constructors